### PR TITLE
enhance: avoid too many gorountines when calc distance

### PR DESCRIPTION
**Pull request by CocaineCong**
use gorountines pool to avoid too many gorountines be created.

### CODE REVIEW

**Comment by SimFG** [comment]
@CocaineCong thanks your contribution. you need to have a sign for the commit to pass the dco check. try to do:
```
git commit --amend -s
```
and then force to push it

**Comment by CocaineCong** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

File: pkg/util/distance/calc_distance.go [hunk 1]
```diff
@@ -130,15 +131,23 @@ func CalcFloatDistance(dim int64, left, right []float32, metric string) ([]float
 
 	distArray := make([]float32, leftNum*rightNum)
 
-	// Multi-threads to calculate distance. TODO: avoid too many go routines
+	// Multi-threads to calculate distance.
 	var waitGroup sync.WaitGroup
-	CalcWorker := func(index int64) {
+	poolSize := int(leftNum / 3)
+	pool, err := ants.NewPoolWithFunc(poolSize, func(i interface{}) {
```
Comment (czs007) [inline pkg/util/distance/calc_distance.go:137]: pool.Release() is not been called?

Comment (czs007) [reply pkg/util/distance/calc_distance.go:137]: Would it be acceptable to continue using the previous approach, but submit a group of coroutines in batches?

Comment (CocaineCong) [reply pkg/util/distance/calc_distance.go:137]: ok, i am accept and thanks for your review.

Comment (chasingegg) [reply pkg/util/distance/calc_distance.go:137]: @CocaineCong could you reopen or submit a new pr by the suggestion above?

Comment (CocaineCong) [reply pkg/util/distance/calc_distance.go:137]: ok，i will resubmit this pr later！

**Pull request closed without merging**